### DISCUSSION TITLE
Wizardcoder-15B support?

### DISCUSSION THREAD

**Discussion by KPTK** [turboderp/exllama#90] [General]
Im getting an error when trying to run Wizardcoder-15B. Is support for this model possible or am I missing something? My understanding is that it is a fine-tune of Starcoder. Is there anything we can do to help Turboderp get support for this as this is a pretty groundbreaking model.  I'll start of gofundme if necessary!

**Comment by turboderp** [comment]
Support is definitely possible, but when new models come out literally every day there's no conceivable way I could support them all. I'm not a team, I'm just a dude, and I have a day job as well.

There's already software that does what you're after, and there's a reason why it's so slow despite having thousands of contributors working on it for years. It's not that those guys don't know what they're doing. They have all the talent, experience and education between them that you could want. But if you're writing a plug-and-play framework that aims to do everything for everyone, what you end up with is Transformers, with all of its various shortcomings.

Now obviously I want to build out this project over time. Llama is kind of old already and it's going to be supplanted at some point. But if I simply jumped on whatever looked promising all the time, I'd have already started adding support for MPT, then stopped halfway through to switch to Falcon instead, then left that in an unfinished state to start working on Starcoder. And of course I wouldn't finish that either because of any number of "groundbreaking" new models released this morning.

The only thing that makes sense for me is to wait and see what catches on. But in the meantime there's nothing stopping anyone from making a Starcoder fork. Then `git clone` and start coding. I've tried to make the code much more approachable than the original GPTQ code I had to work with when I started.
